text/metrics/0113-exemplars.md

```diff
@@ -0,0 +1,94 @@
+# Integrate Exemplars with Metrics
+
+This OTEP adds exemplar support to aggregations defined in the Metrics SDK.
+
+## Definition
+
+Exemplars are example data points for aggregated data. They provide specific context to otherwise general aggregations. For histogram-type metrics, exemplars are points associated with each bucket in the histogram giving an example of what was aggregated into the bucket. Exemplars are augmented beyond just measurements with references to the sampled trace where the measurement was recorded and labels that were attached to the measurement.
+
+## Motivation
+
+Defining exemplar behaviour for aggregations allows OpenTelemetry to support exemplars in Google Cloud Monitoring.
+
+Exemplars provide a link between metrics and traces. Consider a user using a Histogram aggregation to track response latencies over time for a high QPS server. The histogram is composed of buckets based on the speed of the request, for example, "there were 55 requests that took 400-500 milliseconds". The user wants to troubleshoot slow requests, so they would need to find a trace where the latency was high. With exemplars, the user is able to get an exemplar trace from a high latency bucket, an exemplar trace from a low latency bucket, and compare them to figure out the reason for the high latency.
+
+Exemplars are meaningful for all aggregations where relevant traces can provide more context to the aggregation, as well as when exemplars can display specific information not otherwise shown in the aggregation (for example, the full set of labels where they otherwise might be aggregated away).
+
+## Internal details
+
+An exemplar is defined as:
+
+```
+message Exemplar {
+  // Numerical value of the measurement that was recorded. Only one of these two fields is
+  // used for the data, depending on its type
+  double double_value = 0;
+  int64 int64_value = 1;
+  
+  // Exact time that the measurement was recorded
+  fixed64 time_unix_nano = 2;
+
+  // 'label:value' map of all labels that were provided by the user recording the measurement
+  repeated opentelemetry.proto.common.v1.StringKeyValue labels = 3;
+
+  // Span ID of the current trace [Optional]
+  string span_id = 4;
+
+  // Trace ID of the current trace [Optional]
+  string trace_id = 5;
+}
```

Comment:
For me what's most important is that:
(1) exemplars carry an optional statistical weight
(2) exemplars can be included alongside the output of any aggregation
(3) multiple exemplars are supported.

Today we have

```
message Metric {
  // metric_descriptor describes the Metric.
  MetricDescriptor metric_descriptor = 1;

  // Data is a list of one or more DataPoints for a single metric. Only one of the
  // following fields is used for the data, depending on the type of the metric defined
  // by MetricDescriptor.type field.
  repeated Int64DataPoint int64_data_points = 2;
  repeated DoubleDataPoint double_data_points = 3;
  repeated HistogramDataPoint histogram_data_points = 4;
  repeated SummaryDataPoint summary_data_points = 5;
}
```

I'm suggesting that exemplars should be an optional repeated field alongside the other 4 value types.

Reply:
I should have added: this "statistical weight" should be expressed as an adjusted count.  
```
   // When sample_count is non-zero, this exemplar has been chosen in a statistically
   // unbiased way such that the exemplar is representative of `sample_count` individual events.
   optional double sample_count = 10;
```

Reply:
(The "non-zero" statement above allows non-probabilistically selected exemplars to be included.)

Reply:
Added the sample_count parameter, and mentioned that an 
```
optional repeated RawValue exemplars = 6
``` 
parameter would be included on the metric_descriptor object. I wasn't sure about that originally because I was thinking in terms of the client SDK implementation (ie exemplars are computed in the aggregator, so how can it be pulled out?) and not in terms of the proto. It makes sense now, and I agree with it. 

I also revised the wording of the aggregations to allow for statistical sampling, and moved the 'preference for traces' out into a parameter.